numba/targets/optional.py

```diff
@@ -93,6 +93,17 @@ def any_to_optional(context, builder, fromty, toty, val):
         return context.make_optional_value(builder, toty.type, val)
 
 
+@lower_cast(types.Optional, types.boolean)
```

Comment:
Yes, this is needed.  In fact, it is originally there in the lines below.  However, I forgot to remove the old definition, which may raise TypeError.  The new definition allows optional to cast to boolean without raising.